_posts/2021-11-07-canary.md

```diff
@@ -0,0 +1,123 @@
+I1107 15:13:28.421214       1 producer.go:100] Message sent: partition=2, offset=2252, duration=253 ms
+...
+```
+
+Other than the metrics endpoint that is really useful to be scraped by a Prometheus server, there is a `/status` endpoint which exposes some aggregated data from such a metrics in JSON format; it can be useful for some automation where you don't have the ability to scrape and decoding the text Prometheus format but your tool can easily understand JSON text.
+Right now the only data provided through this endpoint is the percentage of messages that are produced and consumed successfully by the canary over a specific time window.
+
+```json
+{
+  "Consuming": {
+    "TimeWindow": 300000,
+    "Percentage": 100
+  }
+}
+```
+
+Everything described above is totally configurable using several environment variables, from the period at which messages are exchanged to the periodic connection check, from the histogram latency buckets to the time window for status data.
+Other than that, the canary supports configuration for connecting to a cluster where TLS is enabled as well as different SASL authentication mechanisms like PLAIN and SCRAM-SHA.
+You can get more information in the ["Configuration"]((https://github.com/strimzi/strimzi-canary#configuration)) section of the project landing page.
+
+## Allow the canary to fly in the mine
+
+The canary tool is not integrated within the Strimzi operator (yet); it could be a future improvement on the project in order to deploy it via a `Kafka` custom resource configuration.
+Right now the release related installation files provide all you need to run the canary on a Kubernetes cluster alongside your Kafka cluster.
+A `Deployment` file is available with a basic configuration to connect to a Kafka cluster without TLS support and exchanging messages every 10 seconds; the other main parameter needed to be changed by the user is about the bootstrap servers address.
+
+```yaml
+env:
+  - name: KAFKA_BOOTSTRAP_SERVERS
+    value: my-cluster-kafka-bootstrap:9092
+  - name: RECONCILE_INTERVAL_MS
+    value: "10000"
+  - name: TLS_ENABLED
+    value: "false"
+```
+
+Together with the deployment, the installation provides a `Service` for exposing the 8080 port which is used for reaching the HTTP endpoints corresponding to metrics, status as well as liveness and readiness.
+
+## Conclusion
+
+Monitoring a Kafka cluster by verifying that it's behaving and working properly is really simple thanks to the canary tool.
+Even when your cluster is not heavily used, it helps to check that everything is working fine to deal with the time when the real applications will start to use it.
+It is also useful to generate some traffic for validating A/B deployment and upgrades.
+
+The canary is just borned with its first [release](https://github.com/strimzi/strimzi-canary/releases), so it is important for us to hear from the community about questions, issues but mostly improvements with new ideas!
```

Comment:
`borned` -> `born`. I wonder if we should drop the first sentence as this will need updating as soon as we make another release.

Reply:
Well as any blog post it makes sense at the date you are writing it.